acstools/utils_findsat_mrt.py

```diff
@@ -0,0 +1,1253 @@
+            print(np.shape(result))
+            radon_image = result[:, 0, :]
+            lengths = result[:, 1, :]
+        radon_image = radon_image.T
+        lengths = lengths.T
+        p.close()
+
+    if print_calc_times:
+        LOG.info('Total time to warp images = {} seconds'.
+                 format(total_warp_time))
+        LOG.info('Total time to calculate medians = {} seconds'.
+                 format(total_median_time))
+
+    if return_length is True:
+        return radon_image, lengths
+    else:
+        return radon_image
+
+
+def create_mrt_line_kernel(width, sigma, outfile=None, shape=(1024, 2048),
+                           plot=False, theta=np.arange(0, 180, 0.5),
+                           threads=1):
+    '''
+    Creates a model signal MRT signal of a line of specified width and blurred
+    by a psf. Used for detection of real linear signals in imaging data.
+
+    Parameters
+    ----------
+    width : int
+        Width of the line. Intensity is constant over this width.
+    sigma : float
+        Gaussian sigma of the PSF. This is NOT FWHM.
+    outfile : string, optional
+        Location to save an output fits file of the kernel. The default is
+        None.
+    sz : tuple/int, optional
+        Size of the image on which to place the line. The default is
+        (1024,2048).
+    plot : bool, optional
+        Flag to plot the original image, MRT, and kernel cutout
+    theta : array, optional
+        Set of angles at which to calculate the MRT, default is
+        np.arange(0,180,0,5)
+    threads: int, optional
+        Number of threads to use when calculating MRT. Default is 1.
+    Returns
+    -------
+    kernel : ndarray
+        The resulting kernel
+
+    '''
+
+    # set up empty image and coordinates
+    image = np.zeros(shape)
+    y0 = image.shape[0]/2-0.5
+    x0 = image.shape[1]/2-0.5
+    xarr = np.arange(image.shape[1])-x0
+    yarr = np.arange(image.shape[0])-y0
+    x, y = np.meshgrid(xarr, yarr)
+
+    # add a simple streak across the image.
+    image = add_streak(image, width, 1, rho=0, theta=90, psf_sigma=sigma)
+
+    # plot the image
+    if (plot is True) & (plt is not None):
+        fig, ax = plt.subplots(figsize=(20, 10))
+        ax.imshow(image, origin='lower')
+        ax.set_xlabel('X')
+        ax.set_ylabel('Y')
+        ax.set_title('model image')
+
+    # calculate the RT for this model
+    rt = radon(image, circle=False, median=True, fill_value=np.nan,
+               threads=threads, return_length=False)
+
+    # plot the RT
+    if (plot is True) & (plt is not None):
+        fig2, ax2 = plt.subplots()
+        ax2.imshow(rt, aspect='auto', origin='lower')
+        ax2.set_xlabel('angle pixel')
+        ax2.set_ylabel('offset pixel')
+
+    # find the center of the signal by summing along each direction and finding
+    # the max.
+    rt_rho = np.nansum(rt, axis=1)
+    rt_theta = np.nansum(rt, axis=0)
+    fig, [ax1, ax2] = plt.subplots(1, 2)
```

Comment:
Won't the rest of this function crash horribly from here onwards if you don't have matplotlib?

Reply:
Yes, thanks for catching that. Those commands should be inside an if statement